City-of-Helsinki/helsinki-paatokset
# Pp 291 search tuning

Add rest of the specified changes to search. 

Prerequisities:
- Make sure you have this variable in your .env.local-file: `REACT_APP_ELASTIC_URL=https://elastic-helsinki-paatokset.docker.so`. If adding it now, use `printenv` inside container to confirm.
- You need to have data from decisions, policymakers and trustees integrations.
- **!** Search application seems to run into some CORS issues in **Firefox** in dev environment. Please use some other browsers to test the feature.

To test:
- `make shell` into the container. To make sure no local files of old versions linger and new ones are installed, run `rm -rf vendor/paatokset && rm -rf public/modules/custom/paatokset_search/assets/* && composer update paatokset/paatokset_search --prefer-source`
- Since we are changing some index settings, send DELETE request to `localhost:9200/paatokset_decisions` and `localhost:9200/paatokset_policymakers` - if you have the indices on your local. You should get a message back with `"acknowledged": true`.
- Either run `make new` or `drush cr; drush cim -y`.
- Check that your indices are back up with a GET request to `localhost:9200/_cat/indices`. You should see both of the indices listed in the response.
- At this point, go and edit a policymaker, eg. `https://helsinki-paatokset.docker.so/fi/paattajat/pormestari`. Add values to the first and last name fields.
- Go to `https://helsinki-paatokset.docker.so/fi/admin/config/search/search-api`. Navigate to each index from the list and index the data. (Index now-button should work usually, but if the index status is 0/0 indexed for some reason, rebuilding the tracking information should solve it).
- After the data is indexed, both search applications should work (https://helsinki-paatokset.docker.so/fi/asia and https://helsinki-paatokset.docker.so/fi/paatokset).

Note that you might need to ctrl+shift+r (or cmd+shift+r) to not get tricked by browser cache.

Things to test for in both search apps:
- Pagination has been fixed to look more like the sketch -> if there are a lot of pages of results, it always shows the last and the first page (something like this: 1... 4 5 6 ... 88)
- Search results don't render unless search is executed OR one of the search params OR pagination param is present in the URL.
- Wildcard queries work and return results. This affects mostly compound words. Eg. searching for 'lääkäri' should return matches for 'ylilääkäri', 'valtuusto' should return matches for 'kaupunginvaltuusto', etc.

Decisions-app changes to test:
- Highlight colors work and are correct
- Decisionmaker title is rendered correctly

Policymaker-app changes to test:
- Remember the policymaker that you edited to include first and last name? You should get that policymaker as a result when searching with those names.
- Highlight colors should work and be correct.
- App width should be in line with other elements on page in < desktop screen sizes.
